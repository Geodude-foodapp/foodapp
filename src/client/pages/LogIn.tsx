import React from 'react';
import axios from 'axios';
import UserForm from '../components/UserForm';
import { UserFormState } from '../../Types';
import { useNavigate } from 'react-router-dom';

export default () => {
  const logIn = (formData: UserFormState) => {
    axios
      .post('/api/login', formData)
      .then((res) => {
        console.log(res);
      })
      .catch((err) => {
        console.error(err);
      });
  };
  const navigate = useNavigate()

  return (
    <section id='login'>
      <UserForm
        type='Log In'
        handleSubmit={logIn}
      />
    </section>
  );
};
