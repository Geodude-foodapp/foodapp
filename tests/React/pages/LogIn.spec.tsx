import { render } from '@testing-library/react';
import React from 'react';
import LogIn from '../../../src/client/pages/LogIn';

describe('LogIn Page', () => {
  beforeAll(() => render(<LogIn />));
  it('matches the snapshot', () => {
    const loginPage = document.querySelector('#login');
    expect(loginPage).toMatchSnapshot();
  });
});
