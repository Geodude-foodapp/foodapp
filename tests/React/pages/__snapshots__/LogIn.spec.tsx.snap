// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn Page matches the snapshot 1`] = `
<section
  id="login"
>
  <section
    id="user-form"
  >
    <h2>
      Log In Form
    </h2>
    <form>
      <input
        placeholder="username"
        type="text"
        value=""
      />
      <input
        placeholder="password"
        type="password"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </section>
</section>
`;
